sale_channel_mirakl/models/sale_channel_mirakl.py

```diff
@@ -0,0 +1,322 @@
+import base64
+import csv
+import hashlib
+import io
+import logging
+from collections import OrderedDict
+from datetime import date, datetime, timedelta
+
+import requests
+
+from odoo import fields, models
+
+from ..mirakl_mapper.mirakl_catalog import MiraklCatalog
+from ..mirakl_mapper.mirakl_offer import MiraklOffer
+from ..mirakl_mapper.mirakl_product import MiraklProduct
+
+_logger = logging.getLogger(__name__)
+DELIMITER = ";"
+MIRAKL_DELAY = 2
+
+
+class SaleChannelMirakl(models.Model):
+    _name = "sale.channel.mirakl"
+    _inherits = {"sale.channel": "channel_id"}
+    _description = "sale channel from Mirakl"
+
+    channel_id = fields.Many2one(
+        comodel_name="sale.channel", required=True, index=True, ondelete="restrict"
+    )
+
+    location = fields.Char(required=True, help="Url to Mirakl application")
+    api_key = fields.Char(help="WebService API Key")
+    url = fields.Char(store=True)
+
+    offer_filename = fields.Char(string="Offers File Name", required=True)
+    shop_id = fields.Char(string="Shop ID on Mirakl")
+
+    data_to_export = fields.Selection(
+        selection=[
+            ("product", "Product"),
+            ("offer", "Offer"),
+            ("catalog", "Catalog"),
+        ],
+        string="Type of data to Export",
+        required=True,
+    )
+
+    default_country_id = fields.Many2one(
+        "res.country",
+        string="Default Country",
+        help="This country will be mentionned by default on adresses if no one"
+        " is provided by the marketplace",
+    )
+
+    import_orders_from_date = fields.Datetime("Import Orders from Date")
+
+    data_to_import = fields.Selection(
+        selection=[("sale.order", "Sale Order")],
+        default="sale.order",
+        string="Type of data to Import",
+    )
+
+    def _get_url(self):
+        return "{}/api/".format(self.location)
+
+    def _get_file_options(self):
+        """
+        Get additional options for the CSV file based on data's type to export.
+        :return: Additional options for the CSV file.
+        """
+        if self.data_to_export == "offer":
+            return MiraklOffer.get_additional_option_for_file()
+        elif self.data_to_export == "catalog":
+            return MiraklCatalog.get_additional_option_for_file()
+        else:
+            return {}
```

Comment:
Please call the same function on the other Pydantic object. In case of someone inherit it to add some data, it'll be lost